pkg/opt/opt.go

```diff
@@ -26,13 +26,12 @@ type CompileOptions struct {
 }
 
 // MergeOptions will merge the input options.
+// When combining multiple options, the 'WithLogWriter' should be set at the end.
 func MergeOptions(opts ...CompileOptions) CompileOptions {
 	var opt = DefaultCompileOptions()
 	for _, o := range opts {
 		opt.Merge(*o.Option)
-		if o.writer != nil {
-			opt.writer = o.writer
-		}
+		opt.writer = o.writer
```

Comment:
This line of code will lead to non idempotence of options, meaning that different options will affect the final option result due to different order settings. It is recommended to adopt builder mode here.

An implementation reference: https://github.com/helm/helm/blob/main/pkg/getter/getter.go#L53